server/db/config.js

```diff
@@ -0,0 +1,87 @@
+const Promise = require('bluebird');
+
+module.exports = (db) => {
+  if (!db.queryAsync) {
+    db = Promise.promisifyAll(db);
+  }
+  // Create products table
+  return db.queryAsync(`
+    CREATE TABLE IF NOT EXISTS products (
+      id INTEGER AUTO_INCREMENT NOT NULL,
+      name VARCHAR(48) NOT NULL ,
+      PRIMARY KEY (id)
+    );`)
+    .then(() => {
+      // Create reviews table
+      return db.queryAsync(`
+      CREATE TABLE IF NOT EXISTS reviews(
+        id INTEGER AUTO_INCREMENT,
+        stars INTEGER(1) NOT NULL,
+        title VARCHAR(60) NOT NULL,
+        body VARCHAR(2000) NOT NULL,
+        recommended INTEGER(1) NOT NULL,
+        location VARCHAR(40) NOT NULL,
+        gift INTEGER(1) NOT NULL,
+        email VARCHAR(40) NOT NULL,
+        pros VARCHAR(48) NULL,
+        cons VARCHAR(48) NOT NULL,
+        describe_yourself VARCHAR(48) NULL,
+        best_uses VARCHAR(48) NULL,
+        product_ID INTEGER NOT NULL,
+        PRIMARY KEY(id)
+        );`);
+    })
+    .then(() => {
+      // create foreign keys
+      return db.queryAsync(`
+      ALTER TABLE reviews ADD FOREIGN KEY(product_ID) REFERENCES products(id);
+        `);
+    }).then(() => {
+      for (var i = 1; i < 101; i++) {
+        db.queryAsync(`INSERT IGNORE INTO products (name) VALUES ('Product Number ${i}')`);
+      }
+    }).then(() => {
+      const dummyText = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.";
+
+      const randomNumber =(min, max) => {
+        min = Math.ceil(min);
+        max = Math.floor(max);
+        return Math.floor(Math.random() * (max - min + 1)) + min; 
+      }
+      const optData = () => {
+        let num = randomNumber(0,1);
+        if (num === 1) {
+          return 'Lorem ipsum dolor sit amet';
+        } 
+        if (num === 0) {
+          return '';
+        };
+      }
+      for (var i = 1; i < 101; i++) {
+        for (var j = 0; j < 5; j++) {
```

Comment:
It might be worth having the second for-loop iterate over a random number between 0 and 5 or 10 — that way you'd get to model what the product reviews app will look like for a product that has no reviews, or a lot of reviews, and everything in between.

Reply:
I can add a random int for reviews, thats a good idea. but i dont understand the edit to the loop. it needs to be 1-101 because its referencing the product id which starts at 1

Reply:
Oh I meant the second loop on `j` from 0–5..

So it could be something like:

```
for (var j = 0; j < randomNumberBetweenZeroAndWhatever; j++) {
```